PR for SRVKE-1742: Document "Event Transformation" section for OpenShift Serverless Eventing (#98334)
openshift/openshift-docs: modules/serverless-event-transformation-deployment-patterns.adoc

// Module included in the following assemblies:
//
// serverless/eventing/serverless-event-transformation.adoc

:_mod-docs-content-type: CONCEPT
[id="serverless-event-transformation-deployment-patterns_{context}"]

> = Deployment patterns for event transformation

Author: I totally agree, but according to the Red Hat style guide and OpenShift doc standard, we can not use `-->` in the titles, so that is why I had to modify the section titles.

> = Deployment patterns for event transformation

Reviewer: That is OK, but than we should have some text that makes the flow obvious - b/c currently for me that is not the case here

> = Deployment patterns for event transformation

Author: I have updated and refined the description.

You can use `EventTransform` at different points in your event flow depending on your architecture needs. The following patterns are supported:

[id="serverless-event-transformation-source-to-broker_{context}"]
== Source to Broker event transformation

You can apply an `EventTransform` resource between a Source and a Broker to modify events before they are delivered to the Broker. This pattern ensures that only normalized or enriched events enter the Broker for downstream processing. 

> You can transform events emitted by an `ApiServerSource` before sending them to the default Broker as shown in the following example: 

Reviewer: Perhaps excplicitly saying that the are routed to the EventTransformer, which than transforms and routes them to the broker?

Author: I will make the same here as well.

Author: I have updated and refined the description.

[source,yaml]
----
apiVersion: eventing.knative.dev/v1
kind: Trigger
metadata:
  name: transform-trigger
spec:
  broker: default
  filter:
    attributes:
      type: original.event.type
  subscriber:
    ref:
      apiVersion: eventing.knative.dev/v1alpha1
      kind: EventTransform
      name: event-transformer
---
apiVersion: eventing.knative.dev/v1alpha1
kind: EventTransform
metadata:
  name: event-transformer
spec:
  sink:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: destination-service
  jsonata:
    expression: |
      # transformation expression
----

[id="serverless-event-transformation-broker-reply_{context}"]
== Broker reply event transformation

You can configure `EventTransform` without a sink so that it republishes transformed events back to the Broker. This pattern is useful when you want transformed events to be routed to additional Triggers or consumers.

[NOTE]
====
When using the Broker reply feature, ensure the transformed events do not match the same Trigger that invoked the `EventTransform`. Otherwise, you risk creating an infinite event loop.
====

> In the following example, events of type `original.event.type` are filtered by a Trigger, transformed by the `EventTransform`, and then republished to the Broker with a new type `transformed.event.type`. When you update the type or another attribute, the event is routed to other Triggers without being re-processed by the same Trigger.

Reviewer 2: ```suggestion
You can filter events of type `original.event.type` with a Trigger, transform them with `EventTransform`, and republish them to the Broker as `transformed.event.type` type. Updating the type or another attribute routes the event to different Triggers without reprocessing by the same Trigger.
```
Suggestion to avoid self-referential language and using noun identifier.
